src/librustc_trans/trans/_match.rs

```diff
@@ -1624,21 +1624,9 @@ pub fn store_arg<'blk, 'tcx>(mut bcx: Block<'blk, 'tcx>,
         Some(ident) => {
             // Generate nicer LLVM for the common case of fn a pattern
             // like `x: T`
```

Comment:
I understood this to be about the Some/None arms in the surrouding match, simple pattern vs. generic pattern.